mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -877,15 +969,25 @@ ArrayAttr mlir::rock::collapseContiguousMerges(ArrayAttr transforms,
       ops.push_back(newMerge);
       changed = true;
     }
+    TransformMapAttr newMap = map;
     if (changed) {
-      auto newMap = TransformMapAttr::get(ops, map.getUpperBounds(),
-                                          map.getLowerBounds());
-      newTransformMaps.push_back(newMap);
+      newMap = TransformMapAttr::get(ops, map.getUpperBounds(),
+                                     map.getLowerBounds());
+      // From now on, we have to copy the transform stack.
+      needCloning = true;
```

Comment:
[Clarification] Why do we started to need to clone *now* ?

Reply:
I'll update, but the point is we don't want to mutate the IR in place because there might be other users of the transform stack.

Though ... yeah, we could probably just replace one singular use ... yeah, now that I think about it, that's viable